src/components/button-toggle/button-toggle.pw.tsx

```diff
@@ -0,0 +1,863 @@
+
+    await expect(buttonToggleButton(page).first()).toHaveAttribute(
+      "aria-label",
+      "cypress-aria"
+    );
+  });
+
+  test("should render with aria-labelledby prop", async ({ mount, page }) => {
+    await mount(
+      <ButtonToggleComponent aria-labelledby={CHARACTERS.STANDARD} />
+    );
+
+    await expect(buttonToggleButton(page).first()).toHaveAttribute(
+      "aria-labelledby",
+      CHARACTERS.STANDARD
+    );
+  });
+
+  ([
+    [true, "true"],
+    [false, "false"],
+  ] as [ButtonToggleProps["pressed"], string][]).forEach(
+    ([state, ariaPressed]) => {
+      test(`should render with aria-pressed attribute ${ariaPressed} when pressed prop is ${state}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<ButtonToggleComponent pressed={state} />);
+
+        await expect(buttonToggleButton(page).first()).toHaveAttribute(
+          "aria-pressed",
+          ariaPressed
+        );
+      });
+    }
+  );
+
+  test("should render with data-component prop set to playwright_data", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ButtonToggleComponent data-component={CHARACTERS.STANDARD} />);
+
+    await expect(
+      buttonToggleButton(page).first().locator("..")
+    ).toHaveAttribute("data-component", CHARACTERS.STANDARD);
+  });
+
+  test("should render with data-element prop set to playwright_data", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ButtonToggleComponent data-element={CHARACTERS.STANDARD} />);
+
+    await expect(
+      buttonToggleButton(page).first().locator("..")
+    ).toHaveAttribute("data-element", CHARACTERS.STANDARD);
+  });
+
+  test("should render with data-role prop set to playwright_data", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ButtonToggleComponent data-role={CHARACTERS.STANDARD} />);
+
+    await expect(
+      buttonToggleButton(page).first().locator("..")
+    ).toHaveAttribute("data-role", CHARACTERS.STANDARD);
+  });
+
+  ([
+    [SIZE.SMALL, 32],
+    [SIZE.MEDIUM, 40],
+    [SIZE.LARGE, 48],
+  ] as [ButtonToggleProps["size"], number][]).forEach(([size, height]) => {
+    test(`should render with height ${height} when size prop is ${size}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <ButtonToggleComponent size={size}> {size}</ButtonToggleComponent>
+      );
+
+      const buttonPreview = buttonTogglePreview(page).nth(0);
+      await assertCssValueIsApproximately(buttonPreview, "height", height);
+    });
+  });
+
+  (["add", "share", "tick"] as const).forEach((type) => {
+    test(`should render with ${type} icon`, async ({ mount, page }) => {
+      await mount(
+        <ButtonToggleComponent buttonIcon={type} buttonIconSize="large">
+          {" "}
+          {type}
+        </ButtonToggleComponent>
+      );
+
+      await expect(icon(page).nth(0)).toHaveAttribute("type", type);
+    });
+  });
+
+  ([
+    [SIZE.SMALL, "small"],
+    [SIZE.MEDIUM, "medium"],
```

Comment:
Does a buttonIconSize of "medium" exist? The values for this prop are only meant to be "small" or "large".